### ISSUE TITLE
kv: LoQ tool generates an incorrect error message when executing a CockroachDB development build.

### ISSUE THREAD

**Issue by shailendra-patel** [cockroachdb/cockroach#130166]
On the `drt-large` cluster, we are encountering issues with unavailable ranges due to a loss of quorum. To address this, I attempted to generate a recovery plan using the `cockroach debug` command. However, when executed with a CockroachDB development build binary, the command produces the following error
```shell
#./cockroach debug recover make-plan --host=:26257 --certs-dir certs > plan.json
ERROR: failed to retrieve replica info from cluster: unsupported cluster info version: version is too old, minimum supported 1000024.1, found 24.2-upgrading-to-24.3-step-012
Failed running "debug recover make-plan"

```

Setting environment variable `COCKROACH_TESTING_FORCE_RELEASE_BRANCH=true` and then running the command fixes the issue and a recovery plan is generated.

**Expected behaviour**
The error message should indicate that the cluster is running a development version and suggest setting the environment variable 
`COCKROACH_TESTING_FORCE_RELEASE_BRANCH=true`

**Comment by itsbilal** [comment]
I don't think we should mention the environment variable at all. It's too risky to suggest it without deeper knowledge of the consequences of disabling dev version offsetting.

The error message can instead be similar to the one from pkg/storage where we mention that one side's a dev version without suggesting a course of action:

```
ERROR: server startup failed: failed to start server: failed to create engines: store last used with cockroach non-development version v23.2-2 cannot be opened by development version v1000023.2-2
```
